test/unit/atomic/test-atomic-ref-accessors.cpp

```diff
@@ -0,0 +1,129 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Source file containing tests for atomic accessor methods
+///
+
+#include <RAJA/RAJA.hpp>
+#include "RAJA_gtest.hpp"
+
+template <typename T, typename AtomicPolicy>
+void testAtomicAccessors()
+{
+  // should also work with CUDA
+  T theval = (T)0;
+  T * memaddr = &theval;
+
+  // explicit constructor with memory address
+  RAJA::AtomicRef<T, AtomicPolicy> test1( memaddr );
+
+  // test store method with op()
+  test1.store( (T)19 );
+  ASSERT_EQ( test1, 19 );
+
+  // test assignment operator
+  test1 = (T)23;
+  ASSERT_EQ( test1, 23 );
+
+  // test load method
+  test1 = (T)29;
+  ASSERT_EQ( test1.load(), 29 );
+}
+
+// Pure CUDA test.
+#if defined(RAJA_ENABLE_CUDA)
+template <typename T, typename AtomicPolicy>
+void testAtomicAccessorsCUDA()
```

Comment:
Should this be running tests on the device?

Reply:
Yes, the intent was to run these on the device. Did I not do that? Or, do you prefer these types of tests to be in a separate CUDA directory?

Reply:
I think the tests should be here. One of the reasons for doing the reorg is to avoid PM-specific directories and implementations.